### PR TITLE
TEST: Cleanups

### PR DESCRIPTION
**Pull request by effigies**
@djarecka @robbisg If you're both good with these updates, feel free to merge.

### CODE REVIEW

**Comment by effigies** [comment]
I guess also make sure these pass Travis. :-)

**Comment by effigies** [comment]
@djarecka My Riot client is having trouble connecting. Assuming https://travis-ci.com/nipy/nibabel/builds/147415802 passes, then the new expectation will be that all PRs into `pytest` should pass Travis.

**Comment by robbisg** [comment]
I'm ok, but I will wait @djarecka.
Then I rebase my personal branch, merge and push other modules!

**Comment by effigies** [comment]
@robbisg Feel free to work off of the `pytest` branch for now, rather than wait for tests. If you avoid the files I've touched, then there shouldn't be any conflict.

**Comment by djarecka** [comment]
sorry, was also having lunch and waiting for my riot to recover, but it looks it's not only me... maybe we should move somewhere once @effigies comes back

File: nibabel/streamlines/tests/test_array_sequence.py [hunk 1]
```diff
@@ -9,8 +9,6 @@
 from ...testing_pytest import assert_arrays_equal
```
Comment (djarecka) [inline nibabel/streamlines/tests/test_array_sequence.py]: this should be removed

**Comment by djarecka** [comment]
two comments: 
- for some reason it was often the case that functions like `assert_array_equal` or `assert_almost_equal` were import from `..testing` (currently `..testing_pytest`) and directly from `nipype.testing`. Let's remove the imports of the numpy function from `testing_pytest` and just keep the direct imports

- do we want to keep `assert_array_equal`? This is relatively easy to convert to standard python function, e.g. `assert (arr1 == arr2).all()`

File: nibabel/streamlines/tests/test_array_sequence.py (hunk 1 above)
Comment (effigies) [reply nibabel/streamlines/tests/test_array_sequence.py]: This is used. Note array**s**.

Comment (djarecka) [reply nibabel/streamlines/tests/test_array_sequence.py]: that's right

**Comment by effigies** [comment]
@djarecka 

> * for some reason it was often the case that functions like `assert_array_equal` or `assert_almost_equal` were import from `..testing` (currently `..testing_pytest`) and directly from `nipype.testing`

One advantage to importing things from `nibabel.testing` instead of `numpy.testing` is that, if numpy changes their API, we can just fix it in one place. But if you feel like homogenizing, please open a PR.

> * do we want to keep `assert_array_equal`? This is relatively easy to convert to standard python function, e.g. `assert (arr1 == arr2).all()`

It's not quite. From [the docs](https://docs.scipy.org/doc/numpy/reference/generated/numpy.testing.assert_array_equal.html)

> In contrast to the standard usage in numpy, NaNs are compared like numbers, no assertion is raised if both objects have NaNs in the same positions.

I fixed the `optpkg` test.

I also made some style changes to `test_analyze`, since I was going to have to run the CI again anyway. Would you care to have another review?

**Comment by djarecka** [comment]
@effigies - so you want to use `assert_array_equal`?  I don't like comparing NaNs, but if this is important, it's fine with me
I understood that you want to remove as much as we can and I was moving all `assert_array_equal` to `().all`, but can revert.

**Comment by djarecka** [comment]
@effigies - sure, importing from `..testing` have advantages, but I  saw test files that imported the same function from both places, what is confusing

**Comment by effigies** [comment]
Oh, sorry. Yeah, I meant things like numpy decorators that don't do anything numpy-specific. For testing arrays, I'm fine sticking with the `numpy.testing.assert_*`. If they're obviously not going to have `nan`s, then switching to `assert np.all(x == y)` or `assert np.array_equal(a, b)` seem fine.

Also, if they survive the translation, then clearly they were fine, so I wouldn't revert unless tests start failing.

> but I saw test files that imported the same function from both places, what is confusing

Can you show an example? I've seen what I thought were duplicates, but one had a plural "arrays".

**Comment by djarecka** [comment]
> Oh, sorry. Yeah, I meant things like numpy decorators that don't do anything numpy-specific. For testing arrays, I'm fine sticking with the `numpy.testing.assert_*`. If they're obviously not going to have `nan`s, then switching to `assert np.all(x == y)` or `assert np.array_equal(a, b)` seem fine.
> 
> Also, if they survive the translation, then clearly they were fine, so I wouldn't revert unless tests start failing.

Ok, I will not revert the changes, but will also not convert `array_*` anymore.


> > but I saw test files that imported the same function from both places, what is confusing
> 
> Can you show an example? I've seen what I thought were duplicates, but one had a plural "arrays".

Can't find it now, maybe I was wrong that were in the same file, but definitely in some files functions   were imported from `nipype.testing` and `nose.tools` and in some other files, they were imported from `..testing`. Perhaps we can leave it as it is and can come back to this later.

**Comment by djarecka** [comment]
@effigies - this is an example that imports some `numpy.testing` functions directly from the package and some from `..testing`:
https://github.com/nipy/nibabel/blob/master/nibabel/tests/test_trackvis.py#L13

**Comment by effigies** [comment]
Gotcha. In that case, it looks like I imported things that were available in `nibabel.testing` from there, and the other from `numpy` because it wasn't in nibabel.

But if you look at the history, it used to prefer to grab things directly from `nose` or `numpy`, so I'm okay with returning to that approach. It's probably easiest to remember.

**Comment by effigies** [comment]
This one looks like it's going to pass. https://travis-ci.com/nipy/nibabel/builds/147436283

If you're good with it, I'll merge it to remove some jobs from the CI queue.

**Comment by djarecka** [comment]
@effigies - sure, let's merge it. And I will rebase my branch before opening PR

**Pull request merged**